src/main/java/org/wildfly/security/http/HttpServerScopes.java

```diff
@@ -27,6 +27,22 @@
 public interface HttpServerScopes {
 
     /**
+     * Returns {@code true} if the specified scope exists.
+     *
+     * @param scope the scope required.
+     * @return {@code true} if the specified scope exists. Otherwise, {@code false}
+     */
+    boolean exists(Scope scope);
```

Comment:
Actually I was thinking these methods would be better on the scope that is actually returned.

The fact that getScope returns an instance of a class shows that the scope type is supported, then it can be checked to see if it actually exists and create it.

Regarding attachments within a scope this could even be made automatic, i.e. a call to get where the scope does not exist would always return null (we know even if it was created it would still return null) only at the time we set the attachment would it really be needed and could be auto created then,